src/get-version.ts

```diff
@@ -1,20 +1,28 @@
-import {writeFile, mkdir} from 'fs/promises'
-import HTTP from 'http-call'
-import {dirname} from 'path'
+import makeDebug from 'debug'
+import {HTTP} from 'http-call'
+import {mkdir, writeFile} from 'node:fs/promises'
+import {dirname} from 'node:path'
 
-// eslint-disable-next-line max-params
-async function run(name: string, file: string, version: string, registry: string, authorization: string) {
+async function run([name, file, version, registry, authorization]: string[]) {
+  const debug = makeDebug('get-version')
+  debug('name:', name)
+  debug('file:', file)
+  debug('version:', version)
+  debug('registry:', registry)
+  debug('authorization:', authorization)
   const url = [
     registry.replace(/\/+$/, ''), // remove trailing slash
-    name.replace('/', '%2f'),      // scoped packages need escaped separator
+    name.replace('/', '%2f'), // scoped packages need escaped separator
   ].join('/')
   const headers = authorization ? {authorization} : {}
   await mkdir(dirname(file), {recursive: true})
   await writeFile(file, JSON.stringify({current: version, headers})) // touch file with current version to prevent multiple updates
-  const {body} = await HTTP.get<any>(url, {headers, timeout: 5000})
-  await writeFile(file, JSON.stringify({...body['dist-tags'], current: version, authorization}))
+  const {body} = await HTTP.get<{'dist-tags': string[]}>(url, {headers, timeout: 5000})
```

Comment:
does the http.get need to wait on the mkdir/writeFile?

Reply:
I'm not sure - and I don't want to take the risk of changing it, especially since this script is run in a unref'd spawned process. So any performance improvements in there will never be noticed by anyone